qiskit_experiments/framework/json.py

```diff
@@ -457,7 +458,11 @@ def default(self, obj: Any) -> Any:  # pylint: disable=arguments-differ
         if isinstance(obj, bytes):
             return _serialize_bytes(obj)
         if dataclasses.is_dataclass(obj):
-            return _serialize_object(obj, settings=dataclasses.asdict(obj))
```

Comment:
More reason not to use a dataclass for classes with a custom json encoding?

Reply:
I think this is a bug (I should write separate PR). Basically end-user can pass arbitrary value to options, analysis result value, etc... So dataclass of dataclass should be acceptable in principle, unless it's clearly stated as edge case.